### ISSUE TITLE
Function documentation

### ISSUE THREAD

**Issue by tebeka** [mlrun/mlrun#16]
We'd like a bi-directional way to communicate help Python <-> YAML

### Python -> YAML
Parse function docstring and populate YAML from it

### YAML -> Python
Have a way to view help in the function object in Jupyter. Probably populate `__doc__` from the YAML

**Comment by tebeka** [comment]
@yaronha Can you point me to the object you refer to as `function`? is it `FunctionEntrypoint`?

**Comment by tebeka** [comment]
Currently we'll work with what sphinx does (reStructuredText).

We can also look at [napoleon](http://www.sphinx-doc.org/en/master/usage/extensions/napoleon.html#module-sphinx.ext.napoleon) for parsing [numpy](https://numpydoc.readthedocs.io/en/latest/format.html) or [Google](https://google.github.io/styleguide/pyguide.html#Comments) style docstrings

**Comment by tebeka** [comment]
@yaronha said:
> for the doc str also need to accept list of handlers, not just search for decorations

**Comment by tebeka** [comment]
After thinking about parsing the source, I don't think we'll actually run it since it means we need all the dependencies installed just to check the source. Will parse the code and use the AST.

**Comment by yaronha** [comment]
agree, we also need the help text to list outputs (not just params)

**Comment by tebeka** [comment]
> agree, we also need the help text to list outputs (not just params)

It's there already
